nextcord/mixins.py

```diff
@@ -65,7 +73,55 @@ async def pins(self) -> List[Message]:
             The messages that are currently pinned.
         """
 
+        warnings.warn(
+            ".pins is deprecated, use .fetch_pins instead.",
+            stacklevel=2,
+            category=FutureWarning,
+        )
+
         channel = await self._get_channel()
         state = self._state
         data = await state.http.pins_from(channel.id)
         return [state.create_message(channel=channel, data=m) for m in data]
+
+    async def fetch_pins(
+        self, limit: int = 50, before: Optional[SnowflakeTime] = None
+    ) -> AsyncIterator[MessagePin]:
+        """|asynciter|
+
+        Returns an async iterator of all messages that are currently pinned in the channel.
+
+        This requires :attr:`~nextcord.Permissions.read_message_history`, otherwise the result will be empty.
+
+        .. versionadded:: 3.2
+
+        .. note::
+
+            Due to a limitation with the Discord API, the :class:`.MessagePin`
+            objects returned by this method do not contain complete
+            :attr:`.MessagePin.message.reactions` data.
+
+        Parameters
+        ----------
+        limit: :class:`int`
+            The maximum number of messages to retrieve. Defaults to 50.
+        before: Optional[Union[:class:`.abc.Snowflake`, :class:`datetime.datetime`]]
+            The pin before which to retrieve pinned messages.
+        after: Optional[Union[:class:`.abc.Snowflake`, :class:`datetime.datetime`]]
+            The pin after which to retrieve pinned messages.
```

Comment:
There is no `after` parameter and "The pin before which to retrieve pinned messages"? That doesn't sound right.